src/EditorFeatures/Core/Implementation/Diagnostics/DiagnosticsClassificationTaggerProvider.cs

```diff
@@ -57,5 +61,26 @@ internal partial class DiagnosticsClassificationTaggerProvider : AbstractDiagnos
 
         protected internal override ITagSpan<ClassificationTag> CreateTagSpan(bool isLiveUpdate, SnapshotSpan span, DiagnosticData data) =>
             new TagSpan<ClassificationTag>(span, _classificationTag);
+
+        protected internal override ImmutableArray<DiagnosticDataLocation> GetLocationsToTag(DiagnosticData diagnosticData)
+        {
+            using var locationsToTagDisposer = ArrayBuilder<DiagnosticDataLocation>.GetInstance(out var locationsToTag);
+
+            // If there are 'unnecessary' locations specified in the property bag, use those instead of the main diagnostic location.
+            if (diagnosticData.AdditionalLocations != null
+                && diagnosticData.AdditionalLocations.Count > 0
+                && diagnosticData.Properties.TryGetValue(WellKnownDiagnosticTags.Unnecessary, out var unnecessaryIndices))
+            {
+                var additionalLocations = diagnosticData.AdditionalLocations.ToImmutableArray();
+                var indices = JsonConvert.DeserializeObject<IEnumerable<int>>(unnecessaryIndices);
```

Comment:
fyi, .net core has JsonDeserializer.  It can handle things like converting to from an `ImmutableArray<int>`.

Reply:
I thought that was .net core 3.0?  I think we're still .netstandard 2.0, so I wasn't sure I could actually use it

Reply:
Ahhhh.  Quite possibly :)   If you can use it, do so :)   If not, guess you'll have to go something like this route :)

Note: i woudl talk to @jasonmalinowski  on his thoughts on adding this dependency.

Reply:
I guess I could use it as a nuget package - https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-overview - but that's still a dependency I have to add (and maybe have to think about shipping into vs... ).

If only it could all be .net core :(